### ISSUE TITLE
"X does not have a public getter" when running schema migrator

### ISSUE THREAD

**Issue by Reprevise** [IO-Design-Team/hive_ce#53]
**Steps to Reproduce**
1. Create the Freezed type below
2. Run the migrator
3. Error: `ExternalSignInPlatformIdentity.platformSlug does not have a public getter`

Is this a build_runner ordering issue?

**Code sample**
```dart
@HiveType(typeId: HiveTypeIds.localSignInAccount)
@Freezed(toJson: true)
class ExternalSignInPlatformIdentity with _$ExternalSignInPlatformIdentity {
  @JsonSerializable(createToJson: true)
  const factory ExternalSignInPlatformIdentity({
    @HiveField(0) required String platformSlug,
    @HiveField(1) String? platformUserId,
    @HiveField(2) String? platformUserHandle,
    @HiveField(3) String? avatarUrl,
    @HiveField(4) String? countryCode,
    @HiveField(5) String? subject,
  }) = _ExternalSignInPlatformIdentity;

  factory ExternalSignInPlatformIdentity.fromJson(Map<String, Object?> json) =>
      _$ExternalSignInPlatformIdentityFromJson(json);
}
```

**Version**
 - Platform: Windows (Where I'm running the command)
 - Flutter version: 3.24.5
 - Hive version: 2.8.0+2

**Comment by Reprevise** [comment]
@Rexios80 I still have the same error. Is this not something you'd have to update the `hive_ce_generator` for? I noticed the override is for `hive_ce`.

**Comment by Rexios80** [comment]
Whoops you're right it should be 

```yaml
dependency_overrides:
  hive_ce_generator:
    git:
      url: https://github.com/IO-Design-Team/hive_ce
      ref: 11773b890b9ac8df2f30f742565cb4e78665a9f2
      path: hive_generator
```

**Comment by Reprevise** [comment]
Well I'm still getting the same issue after switching the override to the generator.

**Comment by Rexios80** [comment]
Let me try adding a special case for freezed classes

**Comment by Rexios80** [comment]
Turns out there was a bug in the migrator unrelated to freezed. Try this commit hash:

0b8973df033eba13f158da2eaf05adac3491f018

**Comment by Reprevise** [comment]
Seems to work now. Unrelated: how can I customize the generator to only look for files in a specific directly?

I used to have this but now when I run build_runner it says it can't find the builder.
> Configuring `hive_ce_generator:hive_ce_generator` in target `trn_v2:trn_v2` but this is not a known Builder
```yaml
hive_generator:
  generate_for:
    - lib/models/**/*.dart
```

**Comment by Rexios80** [comment]
You can see the names of the builders here: https://github.com/IO-Design-Team/hive_ce/blob/main/hive_generator/build.yaml

To do this for the GenerateAdapters builder, use this key: `hive_ce_generator|hive_adapters_generator`

**Comment by Rexios80** [comment]
The fix for migrating freezed classes was released in hive_ce_generator version 1.8.1
